### PR TITLE
Flask setup

### LINKED ISSUE #14: Set Up Flask
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [ ] New feature (change which adds functionality)
- [x] Chore
## Issue Summary
:fire: Set up Flask, Python and DB <br />
files impacted: <br />
:beetle: If it is a bug or issue fix, describe the issue that needs fixing.
## Resources 
List any resources used when working on the issue (e.g. links to articles, stackoverflow, etc.)
Python Installation Docs: https://docs.python.org/3/using/index.html
General Flask Docs: https://flask.palletsprojects.com/en/1.1.x/
Flask Installation Docs: https://flask.palletsprojects.com/en/1.1.x/installation/
Integrate Postgres Tutorial: https://realpython.com/flask-by-example-part-2-postgres-sqlalchemy-and-alembic/

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [x] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)

## Summary of the change.

🔥 If it is a new feature, describe the feature and the iteration of the project
spec it addresses.

Link to issue: https://github.com/track-basket/trackbasket_BE/issues/14

This closes #14 

🐞 If it is a bug or issue fix, describe the issue that was fixed.

Note any questions or struggles as well.

migrations are not yet in the version directory

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+authors = ["Alexis Dumortier <dumortier.alexis@gmail.com>", "Maria Ronauli mronauli@gmail.com"]
+
+[tool.poetry.dependencies]
+python = "^3.7"
+Flask = "1.0.0"
+Flask-RESTful ="*"
+Flask-SQLAlchemy ="*"
+Flask-Migrate="*"
+Flask-Script="*"
+psycopg2="*"
```
Comment (adumortier) [inline pyproject.toml:14]: uses the latest versions

File: trackbasket_be/app.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from flask import Flask, request
+from flask_restful import Resource, Api
+from flask_sqlalchemy import SQLAlchemy
+from models.volunteer import db, Volunteer
+from resources.volunteer import Volunteer
```
Comment (adumortier) [inline trackbasket_be/app.py:5]: the resource needs to be imported last to avoid conflict with the model

File: trackbasket_be/manage.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from flask_script import Manager
+from flask_migrate import Migrate, MigrateCommand
+from app import app, db
+from models.volunteer import Volunteer
```
Comment (adumortier) [inline trackbasket_be/manage.py:4]: we should have named this `VolunteerModel` to avoid confusion with the resource

File: trackbasket_be/app.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+from flask import Flask, request
+from flask_restful import Resource, Api
+from flask_sqlalchemy import SQLAlchemy
+from models.volunteer import db, Volunteer
```
Comment (mronauli) [inline trackbasket_be/app.py:4]: Do we need this line here since we're not using the Volunteer model?

File: trackbasket_be/models/basemodel.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from flask_sqlalchemy import SQLAlchemy
+
+db = SQLAlchemy()
+
+class BaseModel(db.Model):
+  """Base data model for all objects"""
+  __abstract__ = True
+    # define here __repr__ and json methods or any common method
+    # that you need for all your models
+  def __init__(self, **kargs):
```
Comment (mronauli) [inline trackbasket_be/models/basemodel.py:10]: Why are we passing in **kargs instead of *args?

File: trackbasket_be/manage.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from flask_script import Manager
```
Comment (mronauli) [inline trackbasket_be/manage.py:1]: This seems pretty clutch

**Pull request merged**